src/main/java/bio/terra/service/snapshotbuilder/query/Query.java

```diff
@@ -109,11 +109,12 @@ public String renderSQL(CloudPlatformWrapper platform) {
       sql += " " + having.renderSQL(platform);
     }
 
-    // TODO: DC-836 Implement LIMIT for Azure (TOP N + sql)
-    //  This means passing in the platform to renderSQL
-    //  and refactoring the current TableVariable and tableNameGenerator work
-    if (limit != null) {
-      sql += " LIMIT " + limit;
+    if (limit != null && platform != null) {
+      if (platform.isGcp()) {
+        sql += " LIMIT " + limit;
+      } else if (platform.isAzure()) {
+        sql = "TOP " + limit + " " + sql;
+      }
```

Comment:
do we want to add a default case? or throw an error?

Reply:
if the cloudplatformwrapper is null, I think we won't want to error but just won't include the limit. I'd love for someone to validate this assumption though. If the platform inside the wrapper is null, it will default to GCP.

Reply:
Do we do null checks when using platform elsewhere in the query builder code?

Reply:
Good point - I added a null check with an error message here. This is the only place platform is being used in renderSQL. In the other implementations of renderSQL, I don't think we want to do anything differently if the platform is null because it is unused.